### PR TITLE
Fix TimberHandler so that if the process where the handler is created gets forked, the child process can keep logging

### LINKED ISSUE #14: Timber in multiprocess environments (Celery)
Hello, I have experienced quite a lot of pain making Timber work with [Celery](http://www.celeryproject.org/). 

After quite a bit of debugging I think I've managed to identify which was the issue, and a nasty workaround for celery, but I think better though is required in order to fix this at the ``TimberHandler`` level.

The default parallelization mechanism for celery is to spawn subprocess "workers". Assuming you have configured your logging/handlers in the root process, the child subprocess have access to a pristine copy of an already started thread ``FlushWorker`` and a non ``multiprocessing`` Queue. This causes issues when the child process try to use the logger, as they try to start and already started thead and ``put`` items on a ``Queue`` that is not being watch by any ``FlushWorker``.

Possible solutions include (perhaps):
- Creating the ``FlushWorker`` thread when ``emit`` is called for the first time instead on the ``__init__``.
- Keep  starting the ``FlushWorker`` on the main process (and not the childs), but make sure the queue is shared between them.

**tl;dr** 
TimberHandler doesn't work on the subprocess created by celery.

**nasty workaround**
Celery has a signal called "worker_process_init" that it is invoked every time a child process spawns. By adding a fresh instance of ``TimberHandler`` to the logger you are using, you'll workaround this issue by having an individual ``TimberHandler`` and ``FlushWorker`` thread per subprocess.

```python
import celery.signals
from timber import TimberHandler

@celery.signals.worker_process_init.connect
def worker_process_init(*args, **kwargs):
    logger = get_task_logger('<LOGGERNAME>')
    formatter = logging.Formatter('%(asctime)s - %(message)s')
    th = TimberHandler(
        source_id="<TIMBER_SOURCE_ID>",
        api_key="<TIMBER_API_KEY>",
        buffer_capacity="<TIMBER_BUFFER_CAPACITY>",
        raise_exceptions=True
    )
    th.setFormatter(formatter)
    th.setLevel(logging.INFO)
    logger.addHandler(th)
```

### PR DESCRIPTION
**Pull request by jorgebastida**
In essence, this patch makes the queue in the handler be a ``multiprocessing.Queue`` instead of ``queue.Queue``. This change makes all interactions safe to happen both in different processes and threads. 

Another side effect of dealing with multiple processes is the fact that the Flusher can not/should not run as a thread of each of the childs. The suggested solution only starts the flush thread if we detect the running process is the main one.

Existing tests have been adapted so that they are compatible with the ``multiprocessing.Queue`` interface which ``qsize()`` method is not implemented in all platforms.

Related to this issue I found a couple of interesting articles which support this approach:

> The second problem is that fork() doesn’t actually copy everything. In particular, one thing that fork() doesn’t copy is threads. Any threads running in the parent process do not exist in the child process.
https://codewithoutrules.com/2018/09/04/python-multiprocessing/

> Logging to a single file from multiple processes
https://docs.python.org/3/howto/logging-cookbook.html#logging-to-a-single-file-from-multiple-processes

I have tested this patch and it works-for-me™, but some better though would ideal from the maintainers.

Closes https://github.com/timberio/timber-python/issues/14

### CODE REVIEW

File: tests/test_handler.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def test_handler_creates_pipe_from_args(self, MockWorker):
             buffer_capacity=buffer_capacity,
             flush_interval=flush_interval
         )
-        self.assertEqual(handler.pipe.maxsize, buffer_capacity)
+        self.assertEqual(handler.pipe._maxsize, buffer_capacity)
```
Comment (jorgebastida) [inline tests/test_handler.py:33]: Odd, but ``JoinableQueue`` stores ``maxsize`` in a different place...

**Comment by binarylogic** [comment]
Thanks @jorgebastida this is great! We'll review and get it merged today if it checks out.

**Comment by jorgebastida** [comment]
I'm going to do some further testing in our preprod this weekend to see if all logs are getting flush correctly. Will come back to you on monday.

**Comment by jorgebastida** [comment]
It does look like it works!
